### PR TITLE
(#1082) Replace forbidden API calls

### PR DESCRIPTION
**Pull request by Umbrah**
This is for #1082 
I replaced forbidden API calls in org.cactoos.collection package, updated to-do.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1114](https://github.com/yegor256/cactoos/issues/1114) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1114](https://github.com/yegor256/cactoos/issues/1114) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/8160943f-0713-452e-97e9-fbb23ebdc0f0); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/collection/BehavesAsCollection.java [hunk 1]
```diff
@@ -62,26 +62,52 @@ public BehavesAsCollection(final E item) {
     @SuppressWarnings({ "unchecked", "PMD.ClassCastExceptionWithToArray" })
     public boolean matchesSafely(final Collection<E> col) {
         MatcherAssert.assertThat(
-            col, new IsCollectionContaining<>(new IsEqual<>(this.sample))
+            "Must contain item",
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/collection/BehavesAsCollection.java]: Even though `MatcherAssert.assertThat` is still used, we are in transition to OO/Junit5 `new Assertion` for all the new code added.
Please use `new Assertion<>().affirm()`

File: src/test/java/org/cactoos/collection/FilteredTest.java [hunk 2]
```diff
@@ -54,6 +54,7 @@ public void behavesAsCollection() throws Exception {
     @Test
     public void filterList() {
         MatcherAssert.assertThat(
+            "Length must change",
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/collection/FilteredTest.java]: `new Assertion<>().affirm()`

File: src/test/java/org/cactoos/collection/StickyTest.java [hunk 3]
```diff
@@ -79,6 +83,7 @@ public void decoratesArray() throws Exception {
     @Test
     public void testEmpty() {
         MatcherAssert.assertThat(
+            "Must be empty",
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/collection/StickyTest.java]: `new Assertion<>().affirm()`

File: src/test/java/org/cactoos/collection/TailOfTest.java [hunk 4]
```diff
@@ -44,9 +48,13 @@ public void tailCollection() throws Exception {
                 2,
                 "one", "two", "three", "four"
             ),
-            Matchers.contains(
-                "three",
-                "four"
+            new IsCollectionContaining<>(
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/collection/TailOfTest.java]: `throws Exception` is not needed here

**Comment by Umbrah** [comment]
@ilyakharlamov I replaced all MatcherAssert.assertThat with new Assertion<>().affirm() and removed throws Exception where it is not needed, please check

**Comment by ilyakharlamov** [comment]
@llorllale it's good to merge

File: src/test/java/org/cactoos/collection/HeadOfTest.java [hunk 5]
```diff
@@ -38,18 +42,22 @@
...
+            () -> new HeadOf<>(
                 3,
                 "one", "two", "three", "four"
             ),
-            Matchers.contains(
-                "one",
-                "two",
-                "three"
+            new IsCollectionContaining<>(
+                new AnyOf<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/HeadOfTest.java]: @Umbrah are you sure the logic is preserved here? Why not use `AllOf`?

File: src/test/java/org/cactoos/collection/BehavesAsCollection.java [hunk 6]
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
...
-            col, new IsCollectionWithSize<>(new MatcherOf<>(s -> s > 0))
-        );
-        MatcherAssert.assertThat(
-            new ListOf<>((E[]) col.toArray()),
-            new IsCollectionContaining<>(new IsEqual<>(this.sample))
-        );
+        new Assertion<>(
+            "Must contain item",
+            () -> col,
+            new IsCollectionContaining<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/BehavesAsCollection.java]: @umbrah can't you use `HasValues`?

File: src/test/java/org/cactoos/collection/BehavesAsCollection.java [hunk 7]
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
...
+            "Must not be empty",
+            () -> col,
+            new IsNot<>(
+                new IsEmptyCollection<>()
+            )
+        ).affirm();
+        new Assertion<>(
+            "Size must be more than 0",
+            () -> col,
+            new IsCollectionWithSize<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/BehavesAsCollection.java]: @umbrah `HasSize`

File: src/test/java/org/cactoos/collection/BehavesAsCollection.java [hunk 8]
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
...
+                new MatcherOf<>(s -> s > 0)
+            )
+        ).affirm();
+        new Assertion<>(
+            "Array must contain item",
+            () -> new ListOf<>(
+                (E[]) col.toArray()
+            ),
+            new IsCollectionContaining<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/BehavesAsCollection.java]: @umbrah `HasValues`

File: src/test/java/org/cactoos/collection/BehavesAsCollection.java [hunk 9]
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
...
-            new ListOf<>(array),
-            new IsCollectionContaining<>(new IsEqual<>(this.sample))
-        );
-        MatcherAssert.assertThat(
-            col.containsAll(new ListOf<>(this.sample)), new IsEqual<>(true)
-        );
+        new Assertion<>(
+            "Array from collection must contain item",
+            () -> new ListOf<>(array),
+            new IsCollectionContaining<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/BehavesAsCollection.java]: @umbrah `HasValues`

File: src/test/java/org/cactoos/collection/FilteredTest.java (hunk 2 above)
Comment (llorllale) [reply src/test/java/org/cactoos/collection/FilteredTest.java]: @umbrah message should be "length equal to number of items matching filter"

Comment (llorllale) [reply src/test/java/org/cactoos/collection/FilteredTest.java]: @umbrah actually I don't understand what the difference is between this test and the `size()` test below. I think this one can be deleted

File: src/test/java/org/cactoos/collection/FilteredTest.java [hunk 10]
```diff
@@ -43,68 +43,73 @@
...
+        ).affirm();
     }
 
     @Test
-    public void size() throws Exception {
-        MatcherAssert.assertThat(
-            new Filtered<String>(
+    public void size() {
+        new Assertion<>(
+            "Size must change",
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/FilteredTest.java]: @umbrah message must be "size is equal to number of items matching the filter"

File: src/test/java/org/cactoos/collection/ReversedTest.java [hunk 11]
```diff
@@ -170,40 +178,62 @@ public void testClear() throws Exception {
...
+            "Array must contain element",
+            () -> new Reversed<>(
                 new IterableOf<>(
                     1, 2, 3, 4
                 )
             ).toArray(),
-            Matchers.arrayContaining(4, 3, 2, 1)
-        );
+            new IsArrayContaining<>(
+                new AnyOf<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/ReversedTest.java]: @umbrah `AllOf`?

File: src/test/java/org/cactoos/collection/ReversedTest.java [hunk 12]
```diff
@@ -170,40 +178,62 @@ public void testClear() throws Exception {
...
+            "Array for Reversed must contain elements",
+            () -> new Reversed<>(
                 new IterableOf<>(
                     1, 2, 3, 4, 5
                 )
             ).toArray(new Integer[]{5, 6}),
-            Matchers.arrayContaining(5, 4, 3, 2, 1)
-        );
+            new IsArrayContaining<>(
+                new AnyOf<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/ReversedTest.java]: @umbrah `AllOf`?

Comment (llorllale) [reply src/test/java/org/cactoos/collection/ReversedTest.java]: @Umbrah better yet, `HasValues`

File: src/test/java/org/cactoos/collection/SkippedTest.java [hunk 13]
```diff
@@ -25,103 +25,126 @@
...
             "Can't skip elements in iterable",
-            new Skipped<>(
+            () -> new Skipped<>(
                 2,
                 new IterableOf<>("one", "two", "three", "four")
             ),
-            Matchers.contains(
-                "three",
-                "four"
+            new IsCollectionContaining<>(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/SkippedTest.java]: @umbrah `HasValues`

**Comment by llorllale** [comment]
@umbrah see above.

Basically I worry that the logic may not have translated correctly into the new refactored code. Also, take a look at the available matchers from `cactoos-matchers` - will probably make your life easier in most cases.

**Comment by Umbrah** [comment]
@llorllale The logic was incorrect, I updated all usages of IsCollectionContaining<>(AnyOf<>(...)) with AllOf<>(IsCollectionContaining<>(...)). I cannot use HasValues and HasSize as we do not have it in last published version of cactoos-matchers. Please check.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 35 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#1114](https://github.com/yegor256/cactoos/issues/1114) is not in the agenda of @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), can't inspect

**Comment by 0crat** [comment]
The job [#1114](https://github.com/yegor256/cactoos/issues/1114) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**